src/oracle.ts

```diff
@@ -76,3 +83,66 @@ export async function getOracleDecimals(
     throw new Error(`Failed to fetch oralce decimals: ${result.error}`);
   }
 }
+
+export function addReflectorEntries(tx: Transaction): Transaction {
+  if (tx.toEnvelope().switch() !== xdr.EnvelopeType.envelopeTypeTx()) {
+    return tx;
+  }
+
+  const sorobanData = tx.toEnvelope().v1().tx().ext().sorobanData();
+  const readEntries = sorobanData.resources().footprint().readOnly();
+  const readWriteEntries = sorobanData.resources().footprint().readWrite();
+
+  const newReadEntries = [];
+  for (const entry of readEntries) {
+    if (readWriteEntries.length + readEntries.length + newReadEntries.length + 1 > 100) {
+      break;
+    }
+
+    switch (entry.switch()) {
+      case xdr.LedgerEntryType.contractData(): {
+        const contractData = entry.contractData();
+        const address = Address.fromScAddress(contractData.contract()).toString();
+
+        if (REFLECTOR_ORACLE_ADDRESSES.includes(address)) {
+          switch (contractData.key().switch()) {
+            case xdr.ScValType.scvU128(): {
+              const u128Key = contractData.key().u128();
+              const roundTimestamp = u128Key.hi().toBigInt();
+              const index = u128Key.lo().toBigInt();
+
+              const newRoundTimestamp = roundTimestamp + 300_000n;
+
+              newReadEntries.push(
```

Comment:
this can cause a potentially problematic number of duplicate entries.

Most Blend V2 oracle calls invoke multiple price entries for a single asset index.

IE - assume asset index 1 is called, it would include the entries:

1. price, asset 1, t_now
2. price, asset 1, t_now - (1 * round_period)
3.  price, asset 1,  t_now - (2 * round_period)

In this case, the logic should only include one additional entry,

-> price, asset 1, t_now + round_period